lang/src/org/partiql/lang/eval/io/DelimitedValues.kt

```diff
@@ -131,34 +124,36 @@ object DelimitedValues {
     fun writeTo(ion: IonSystem,
                 output: Writer,
                 value: ExprValue,
-                delimiter: String,
+                delimiter: Char,
                 newline: String,
-                writeHeader: Boolean): Unit {
-        val nullValue = ion.newNull()
+                writeHeader: Boolean) {
+        val csvPrinter = CSVPrinter(output, CSVFormat.DEFAULT.withDelimiter(delimiter).withRecordSeparator(newline))
```

Comment:
Probably aught to be using Kotlin's [use](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) here since your manual invocation of `.close()` below will be skipped if an exception occurs.  (`.use` is Kotlin's equivalent of Java's [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html).)

Reply:
Will change it in the next revision.